### PR TITLE
Login API implementation

### PR DESCRIPTION
**Pull request by camilo05**
Sep-14-2021 02-02-14

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/auth/login/AuthFragment.kt [hunk 1]
```diff
@@ -54,6 +55,17 @@ class AuthFragment @Inject constructor() : WolmoFragment<FragmentLoginBinding, A
...
 
+    override fun showErrorLogin(status: ResponseStatus) {
+        when (status) {
+            ResponseStatus.ERROR_CREDENTIALS -> toastFactory.show(R.string.request_error_login)
+            else -> toastFactory.show(R.string.default_error)
+        }
+    }
+
+    override fun showLoader(visible: Boolean) {
+        with(binding) { loaderLogin.visibility = if (visible) View.VISIBLE else View.INVISIBLE }
```
Comment (LisandroDiMeo) [inline app/src/main/java/ar/com/wolox/android/example/ui/auth/login/AuthFragment.kt]: I think in this cases where you only access to one element of the viewBinding, it's more easy to write `binding.loaderLogin`

File: app/src/main/java/ar/com/wolox/android/example/model/LoginResponse.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package ar.com.wolox.android.example.model
+
+data class LoginResponse(
+    val data: Data,
+    val success: Boolean,
+    val errors: List<String>
+)
+
+data class Data(
```
Comment (LisandroDiMeo) [inline app/src/main/java/ar/com/wolox/android/example/model/LoginResponse.kt]: Perhaps the name `Data` it's not very explanatory. Also, when you are declaring a data class to handle the json object from the request, you can use `@SerializedName("varName")` to reference certain field from the object, thus using your own variable name. Example:
```
data class TransactionModel(
        @SerializedName("idReferencia") val id: String = "",
        @SerializedName("fecha") val date: String = "",
        @SerializedName("detalle") val detail: String = "",
        @SerializedName("importe") val amount: String = "",
)
```

File: app/src/main/java/ar/com/wolox/android/example/ui/auth/login/AuthFragment.kt (hunk 1 above)
Comment (ffasitella) [reply app/src/main/java/ar/com/wolox/android/example/ui/auth/login/AuthFragment.kt]: You can also use the `.togglePresence() `method. It's a View extension function.
`loaderLogin.togglePresence(visible)`

Comment (rodrigo103) [reply app/src/main/java/ar/com/wolox/android/example/ui/auth/login/AuthFragment.kt]: Also, you could replace 
loaderLogin.visibility = if (visible) View.VISIBLE else View.INVISIBLE
by
loaderLogin.toggleVisibility(visible)

**Review by ffasitella — CHANGES_REQUESTED** [review]
Just one comment!

**Review by LisandroDiMeo — APPROVED** [review]
💯

**Review by ffasitella — APPROVED** [review]
Amazing job! Well done! 👍🏽 🥇

**Pull request merged**